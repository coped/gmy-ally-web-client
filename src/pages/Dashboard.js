import React from "react";
import { useAuth } from "context/auth";
import { Button } from "components/common";

export default function Dashboard() {
  const { setAuthTokens } = useAuth();

  function logOut() {
    setAuthTokens("");
  }
  return (
    <div id="Dashboard">
      <p>This is the dashboard.</p>
      <Button onClick={logOut}>Log out</Button>
    </div>
  );
}
