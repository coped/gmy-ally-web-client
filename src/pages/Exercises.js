import React, { useEffect, useState } from "react";
import { Api } from "lib/api";
import { PageLoader } from "components/common";
import { useExercises } from "context/exercises";
import "bulma-quickview/dist/css/bulma-quickview.min.css";
import bulmaQuickview from "bulma-quickview/dist/js/bulma-quickview.min.js";

export default function Exercises() {
  const {
    allExercises,
    muscleGroups,
    categorizedExercises,
    setAllExercises,
  } = useExercises();
  const [loading, setLoading] = useState(false);
  const [currentMuscleGroup, setCurrentMuscleGroup] = useState("abdominals");
  const [quickviewExercise, setQuickviewExercise] = useState();

  // Fetch exercise data on component mount
  useEffect(() => {
    async function fetchData() {
      setLoading(true);
      const data = await Api.indexExercises();
      setAllExercises(data.payload.exercises);
      setQuickviewExercise(data.payload.exercises[0]);
      setLoading(false);
    }
    !allExercises && fetchData();
  }, [allExercises, setAllExercises]);

  useEffect(() => {
    if (allExercises && !quickviewExercise)
      setQuickviewExercise(allExercises[0]);
    if (quickviewExercise) bulmaQuickview.attach();
  });

  function showExercises() {
    console.log(allExercises);
    allExercises && console.log(categorizedExercises);
  }

  function capitalize(string) {
    return string[0].toUpperCase() + string.substring(1);
  }

  return (
    <div id="Exercises">
      <PageLoader loading={loading} className="is-info" />
      {quickviewExercise && (
        <div id="quickviewDefault" className="quickview">
          <header className="quickview-header">
            <p className="title">{quickviewExercise.title}</p>
            <span className="delete" data-dismiss="quickview"></span>
          </header>

          <div className="quickview-body">
            <div className="quickview-block section">
              <h3 className="is-size-5">Movement type:</h3>
              <p>{quickviewExercise.movement_type}</p>
              <h3 className="is-size-5">Equipment:</h3>
              <div className="content">
                <ul>
                  {quickviewExercise.equipment.split("\n").map((item) => (
                    <li key={item}>{capitalize(item)}</li>
                  ))}
                </ul>
              </div>
              <h3 className="is-size-5">Steps:</h3>
              <div className="content">
                <ol>
                  {quickviewExercise.steps.split("\n").map((step, index) => (
                    <li key={"step-" + index}>{step}</li>
                  ))}
                </ol>
              </div>
            </div>
          </div>

          <footer className="quickview-footer">GYM PARTNER</footer>
        </div>
      )}
      <div className="columns">
        <div className="column is-3">
          <aside className="menu">
            <p className="menu-label">SEARCH EXERCISES BY MUSCLE GROUP</p>
            <ul className="menu-list">
              {muscleGroups &&
                muscleGroups.map((name) => (
                  <li key={name}>
                    <a onClick={() => setCurrentMuscleGroup(name)}>
                      {capitalize(name)}
                    </a>
                  </li>
                ))}
            </ul>
          </aside>
        </div>
        <div className="column">
          <h3 className="is-size-3 has-text-centered">
            {capitalize(currentMuscleGroup)}
          </h3>
          <div className="box">
            <ul className="menu-list">
              {categorizedExercises &&
                categorizedExercises[currentMuscleGroup].map((exercise) => (
                  <li className="list-item" key={exercise.name}>
                    <a
                      onClick={() => setQuickviewExercise(exercise)}
                      data-show="quickview"
                      data-target="quickviewDefault"
                    >
                      {exercise.title}
                    </a>
                  </li>
                ))}
            </ul>
          </div>
        </div>
      </div>
      <button onClick={showExercises} className="button">
        Show exercises
      </button>
    </div>
  );
}
