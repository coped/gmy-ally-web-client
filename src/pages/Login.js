import React, { useState } from "react";
import { FormInput, FormButton } from "components/form";
import { AsyncRequest, endpoints, Messages } from "lib";
import { Notification } from "components/common";
import "assets/Login.scss";
import { Link, Redirect } from "react-router-dom";
import { useAuth } from "context/auth";

export default function Login() {
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [apiMessages, setApiMessages] = useState("");
  const [form, setForm] = useState({ email: "", password: "" });

  const { setAuthTokens } = useAuth();

  function onChange(event) {
    const target = event.target;
    setForm({
      ...form,
      [target.name]: target.value,
    });
  }

  function loginUser(credentials) {
    setIsLoading(true);
    AsyncRequest.post(endpoints.authentication.login, credentials)
      .then((response) => response.json())
      .then((data) => {
        if (data.status === "success") {
          console.log(data);
          setAuthTokens(data.payload.jwt);
          setIsLoggedIn(true);
        } else {
          setApiMessages(data.messages);
          setIsLoading(false);
        }
      })
      .catch((error) => {
        console.log(error);
        setApiMessages([Messages.connectionError]);
        setIsLoading(false);
      });
  }

  function onFormSubmit(e) {
    e.preventDefault();
    loginUser({
      login: {
        email: form.email,
        password: form.password,
      },
    });
  }

  if (isLoggedIn) {
    return <Redirect to="/dashboard" />;
  }

  return (
    <div id="Login">
      <h1 className="login-title">Log in</h1>
      <form onSubmit={onFormSubmit} className="column is-6 box">
        {apiMessages &&
          apiMessages.map((response, index) => (
            <Notification key={index} type={response.type}>
              <p>{response.message}</p>
            </Notification>
          ))}
        <FormInput
          id="email-input"
          label="Email:"
          name="email"
          type="email"
          placeholder="your@email.com"
          value={form.email}
          onChange={onChange}
        />
        <FormInput
          id="password-input"
          label="Password:"
          name="password"
          type="password"
          value={form.password}
          onChange={onChange}
        />
        <FormButton classList={["is-link"]} loading={isLoading}>
          Log in
        </FormButton>
        <p>
          Don't have an account? <a href="/signup">Sign up</a>.
        </p>
      </form>
    </div>
  );
}
