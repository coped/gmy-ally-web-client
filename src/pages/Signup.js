import React, { Component } from "react";
import { Notification } from "components/common";
import { FormInput, FormButton } from "components/form";
import { AsyncRequest, endpoints, Messages } from "lib";
import "assets/Signup.scss";

export default class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      form: {
        name: "",
        email: "",
        password: "",
        passwordConfirmation: "",
      },
      apiMessages: "",
    };

    this.onChange = this.onChange.bind(this);
    this.signupUser = this.signupUser.bind(this);
  }

  signupUser(credentials) {
    this.setState({ isLoading: true });
    AsyncRequest.post(endpoints.users.create, credentials)
      .then((response) => response.json())
      .then((data) => {
        if (data.status === "success") {
          this.props.authenticate(data);
        } else {
          this.setState({ apiMessages: data.messages });
        }
      })
      .catch((error) => {
        console.log(error);
        this.setState({ clientMessages: [Messages.connectionError] });
      })
      .finally(() => this.setState({ isLoading: false }));
  }

  onChange(event) {
    const target = event.target;
    this.setState({
      form: {
        ...this.state.form,
        [target.name]: target.value,
      },
    });
  }
  render() {
    const { apiMessages, form } = this.state;
    return (
      <div id="Signup">
        <h1 className="signup-title">Sign up</h1>
        <form onSubmit={this.onFormSubmit} className="column is-6 box">
          {apiMessages &&
            apiMessages.map((message, index) => (
              <Notification key={index} type={message.type}>
                <p>{message.message}</p>
              </Notification>
            ))}
          <FormInput
            id="name-input"
            label="Name:"
            name="name"
            type="text"
            placeholder="Your name"
            value={form.name}
            onChange={this.onChange}
          />
          <FormInput
            id="email-input"
            label="Email:"
            name="email"
            type="email"
            placeholder="your@email.com"
            value={form.email}
            onChange={this.onChange}
          />
          <FormInput
            id="password-input"
            label="Password:"
            name="password"
            type="password"
            value={form.password}
            onChange={this.onChange}
          />
          <FormInput
            id="password-confirmation-input"
            label="Password confirmation:"
            name="passwordConfirmation"
            type="password"
            value={form.passwordConfirmation}
            onChange={this.onChange}
          />
          <FormButton classList={["is-link"]}>Sign up</FormButton>
          <p className="is-size-6">
            Already have an account? <a href="/login">Log in</a>.
          </p>
        </form>
      </div>
    );
  }
}
