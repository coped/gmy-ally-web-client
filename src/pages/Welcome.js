import React from "react";
import "assets/Welcome.scss";
import { Button } from "components/common";

export default function Welcome() {
  return (
    <div id="Welcome">
      <a href="/signup">
        <Button classList={["is-primary"]}>Sign up</Button>
      </a>
      <a href="/login">
        <Button classList={["is-link"]}>Log in</Button>
      </a>
    </div>
  );
}
