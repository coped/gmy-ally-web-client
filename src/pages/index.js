import Login from "pages/Login";
import Signup from "pages/Signup";
import Welcome from "pages/Welcome";
import Dashboard from "pages/Dashboard";
import Exercises from "pages/Exercises";

export { Login, Signup, Welcome, Dashboard, Exercises };
