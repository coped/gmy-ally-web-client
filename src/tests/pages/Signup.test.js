import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Signup } from "pages";

Enzyme.configure({ adapter: new Adapter() });

describe("Signup", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Signup />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders a Signup form", () => {
    const wrapper = shallow(<Signup />);
    expect(wrapper.find("form").length).toEqual(1);
  });

  it("changes to input are reflected in signup state", () => {
    const wrapper = shallow(<Signup />);
    const changeEmail = { target: { name: "email", value: "my-email" } };
    const changePassword = {
      target: { name: "password", value: "my-password" },
    };
    wrapper.find("#email-input").simulate("change", changeEmail);
    wrapper.find("#password-input").simulate("change", changePassword);
    expect(wrapper.state().form.email).toEqual("my-email");
    expect(wrapper.state().form.password).toEqual("my-password");
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<Signup />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});