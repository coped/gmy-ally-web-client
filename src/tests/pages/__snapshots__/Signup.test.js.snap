// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup has a valid snapshot 1`] = `
<div
  id="Signup"
>
  <h1
    className="signup-title"
  >
    Sign up
  </h1>
  <form
    className="column is-6 box"
  >
    
    <div
      className="field"
    >
      <label
        className="label"
      >
        Name:
      </label>
      <div
        className="control"
      >
        <input
          className="input"
          id="name-input"
          name="name"
          onChange={[Function]}
          placeholder="Your name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <label
        className="label"
      >
        Email:
      </label>
      <div
        className="control"
      >
        <input
          className="input"
          id="email-input"
          name="email"
          onChange={[Function]}
          placeholder="your@email.com"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <label
        className="label"
      >
        Password:
      </label>
      <div
        className="control"
      >
        <input
          className="input"
          id="password-input"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <label
        className="label"
      >
        Password confirmation:
      </label>
      <div
        className="control"
      >
        <input
          className="input"
          id="password-confirmation-input"
          name="passwordConfirmation"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control"
      >
        <button
          className="button is-link"
        >
          Sign up
        </button>
      </div>
    </div>
    <p
      className="is-size-6"
    >
      Already have an account? 
      <a
        href="/login"
      >
        Log in
      </a>
      .
    </p>
  </form>
</div>
`;
